src/command/new/mod.rs

```diff
@@ -0,0 +1,348 @@
+use console::Term;
+use dialoguer::Select;
+use saucer::Utf8PathBuf as PathBuf;
+use saucer::{clap, Parser};
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+use std::fmt;
+use std::fs::read_dir;
+use std::io::Write;
+use std::str::FromStr;
+
+use crate::options::TemplateOpt;
+use crate::{anyhow, command::RoverOutput, error::RoverError, Result};
+
+#[derive(Debug, Clone, Serialize, Parser)]
+pub struct New {
+  #[clap(flatten)]
+  options: TemplateOpt,
+
+  /// Path to create template at
+  #[clap(name = "path")]
+  #[serde(skip_serializing)]
+  path: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Parser)]
+pub struct GithubTemplate {
+  pub id: String,
+  pub display: String,
+  pub language: String,
+  pub project_type: String,
+}
+
+impl fmt::Display for GithubTemplate {
+  // This trait requires `fmt` with this exact signature.
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    // Write strictly the first element into the supplied output
+    // stream: `f`. Returns `fmt::Result` which indicates whether the
+    // operation succeeded or failed. Note that `write!` uses syntax which
+    // is very similar to `println!`.
+    write!(f, "{}", self.display)
+  }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Parser)]
+pub enum ProjectType {
+  ALL,
+  SUBGRAPH,
+  CLIENT,
+}
+
+impl PartialEq<String> for ProjectType {
+  fn eq(&self, other: &String) -> bool {
+    match &self {
+      ProjectType::ALL => other.to_lowercase() == "all",
+      ProjectType::SUBGRAPH => other.to_lowercase() == "subgraph",
+      ProjectType::CLIENT => other.to_lowercase() == "client",
+    }
+  }
+}
+impl PartialEq for ProjectType {
+  fn eq(&self, other: &ProjectType) -> bool {
+    match &self {
+      ProjectType::ALL => other.eq(&String::from("all")),
+      ProjectType::SUBGRAPH => other.eq(&String::from("subgraph")),
+      ProjectType::CLIENT => other.eq(&String::from("client")),
+    }
+  }
+}
+impl fmt::Display for ProjectType {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    match self {
+      ProjectType::ALL => write!(f, "all"),
+      ProjectType::SUBGRAPH => write!(f, "subgraph"),
+      ProjectType::CLIENT => write!(f, "client"),
+    }
+  }
+}
+
+impl FromStr for ProjectType {
+  type Err = saucer::Error;
+
+  fn from_str(input: &str) -> std::result::Result<Self, Self::Err> {
+    match input {
+      "all" => Ok(ProjectType::ALL),
+      "subgraph" => Ok(ProjectType::SUBGRAPH),
+      "client" => Ok(ProjectType::CLIENT),
+      _ => Err(anyhow!("Invalid Project Type")),
+    }
+  }
+}
+
+impl New {
+  pub fn run(&self, client_config: StudioClientConfig) -> Result<RoverOutput> {
+    // read_dir will always have the root element, or a count of 1
+    if read_dir(&self.path).unwrap().count() > 1 {
+      return Err(RoverError::new(anyhow!(
+        "You can only create projects in a blank folder. This is to prevent from accidentally overwriting any work."
+      )));
+    }
```

Comment:
we should remove the `unwrap` here because i think it will cause a panic if the directory doesn't exist. if it doesn't exist, we should create it. 

_something_ like this (i have not tested if it compiles)

```suggestion
    if let Ok(dir_entries) = fs::read_dir(&self.path, "") {
      if dir_entries.count() > 1 {
           return Err(RoverError::new(anyhow!(
              "You can only create projects in a blank folder. This is to prevent from accidentally overwriting any work."
            )));
      }
    } else {
      Fs::create_dir_all(&self.path, "")?;
    }
```